### PR TITLE
Deprecate DATADOG* for DD* environment variables in sitecustomize

### PR DESCRIPTION
**Pull request by wantsui**
## Description

Deprecate DATADOG* environment variables in favor of DD*. This doesn't remove the environment variables and only logs a deprecation warning if used.

- For `DATADOG_TRACE_AGENT_HOSTNAME`, use `DD_AGENT_HOST`
- For `DATADOG_TRACE_AGENT_PORT` use `DD_AGENT_PORT`
- For `DATADOG_PRIORITY_SAMPLING`, follow [ingestion controls from the Tracing without Limits configuration](https://docs.datadoghq.com/tracing/trace_retention_and_ingestion/#recommended-setting-the-global-ingestion-rate-to-100).
- For `DATADOG_PATCH_MODULES`, use `DD_PATCH_MODULES`

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/bootstrap/sitecustomize.py [hunk 1]
```diff
@@ -88,7 +90,8 @@ def update_patched_modules():
 
     opts = {}  # type: Dict[str, Any]
 
-    if asbool(os.environ.get("DATADOG_TRACE_ENABLED", True)):
+    dd_trace_enabled = get_env("trace", "enabled")
```
Comment (brettlangdon) [inline ddtrace/bootstrap/sitecustomize.py:93]:
```suggestion
    dd_trace_enabled = get_env("trace", "enabled", default=True)
```

Comment (wantsui) [reply ddtrace/bootstrap/sitecustomize.py]: I'm going to mark this suggestion as resolved as the latest commit I tried does use `dd_trace_enabled = get_env("trace", "enabled", default=True)` and this did fix the error I was getting with some integrations not being patched.

**Review by Kyle-Verhoog — COMMENTED** [review]
Looks good! Just a couple suggestions

File: ddtrace/bootstrap/sitecustomize.py [hunk 2]
```diff
@@ -60,7 +60,7 @@
 
 
 def update_patched_modules():
-    modules_to_patch = os.environ.get("DATADOG_PATCH_MODULES")
+    modules_to_patch = get_env("patch", "modules")
```
[the remark this answers is not in the document]
Comment (Kyle-Verhoog) [reply ddtrace/bootstrap/sitecustomize.py:63]: Good question - this should land in v0.55 (v0.54 should be out later this week).

[the remark this answers is not in the document]
Comment (wantsui) [reply ddtrace/bootstrap/sitecustomize.py:63]: Thanks! The latest commit updates the configuration table with `DATADOG_PATCH_MODULES`!

**Review by Kyle-Verhoog — DISMISSED** [review]
:+1:

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**